### PR TITLE
leap now casts properly

### LINKED ISSUE #13581: Infinite Jump
(WEB REPORT BY: chucklemonger REMOTE: 172.93.109.202:7777)
>Revision (Should be above if you're viewing this from ingame!)
 9c6c0356db96beecb2dc4507668c7e955a5000f2 

>General description of the issue
The Jump genetic power can be repeated endlessly for a very short time.

>What you expected to happen
A single long distance jump.

>What actually happened
Upon being clicked while still in use, it activates again. This can be repeated until the original jump would have hit the ground. When tested with an autoclicker on a 3 millisecond delay, this effectively allows teleportation. Also, it packs the chatbox.

>Steps to reproduce if possible
Acquire genetic Jump power.
Click the icon multiple times in quick succession.

### LINKED ISSUE #15278: Jump major buff-genetics
(WEB REPORT BY: zushakon REMOTE: 172.93.109.202:7777)
>Revision 
d2b45f9c028a58ad5879a8939ba61ed2a8cc4229 
>Jump does not go on CD until around 2 seconds after you press it. Allowing you to jump almost 6 times before the CD is turned on
>What you expected to happen
Jump once

>What actually happened
Jump lots if you spam the shit out of it, covering ground quickly, fucking up your character sprite since you almost go off screen

>Steps to reproduce if possible
get jump genetic buff
Spam jump alot
Jump alot

### PR DESCRIPTION
**Pull request by ghost**
As it depends on a sleep for its fancy animation, it would take some time for cast to finally call take_charge, which would let you stack casts of the jump spell.

Now, everything's handled in after_cast, as cast just had a second argument that was never used anyway.

Also adds some sanity so you're not jumping in space of all places.

Tested, no runtimes that I could see. Couldn't stack-jump or jump into the nth dimension in space.

Fixes #15278
Fixes #13581

### CODE REVIEW

**Comment by SonixApache** [comment]
rip rapid space travel
~~does this fix the thing where you can cut the cooldown by just spamming the button?~~ neat

**Comment by Blithering** [comment]
RIP energy legs

**Comment by kilozombie** [comment]
I dislike this not because I think it's wrong to fix bugs, but because this bug is really cool.
I'll miss it if this ends up getting merged.

**Comment by Exxion** [comment]
>if

**Comment by Fruktlimpa** [comment]
Does this prevent people spamjumping so that they practically leave the screen and all kinds of fucked up shit happens with their sprite and causing lag and all hell? If so, good.

**Comment by ghost** [comment]
Yes. Now each cast only lets you jump once.

**Comment by SonixApache** [comment]
turbojump ability that undoes this WHEN

**Comment by FudgePucker** [comment]
Quantum frog boots that let you jump in space when?

File: code/game/dna/genes/goon_powers.dm [hunk 1]
```diff
@@ -446,6 +446,10 @@
 		var/failed_leap = 0
 		if (istype(target.loc,/turf/))
 
+			if(istype(target.loc, /turf/space))
+				to_chat(target, "<span class='warning'>You're in space. No.</span>")
```
Comment (Intigracy) [inline code/game/dna/genes/goon_powers.dm]: How about something with actual flavor, like "You have nothing to leap off of."

Seperately, why not a check for catwalks?

**Comment by strawberrytan** [comment]
No more leaping through space

RIP

**Comment by Blithering** [comment]
@FudgePucker seven league boots as a new wizard artifact?

**Comment by ghost** [comment]
Boots of blinding speed when?

File: code/game/dna/genes/goon_powers.dm [hunk 2]
```diff
@@ -437,7 +437,7 @@
 	hud_state = "gen_leap"
 	override_base = "genetic"
 
-/spell/targeted/leap/cast(list/targets, mob/user)
+/spell/targeted/leap/after_cast(list/targets)
```
Comment (PJB3005) [inline code/game/dna/genes/goon_powers.dm]: Just use spawn or waitfor don't change this.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Comment by RealestEstate** [comment]
I was actually planning on removing superjumping in space myself, but Skully beat me to it, gg. I'll try to make a pr that makes super farts propel the user in space at crazy velocities. don't you guys fret.

File: code/game/dna/genes/goon_powers.dm (hunk 2 above)
Comment (gbasood) [reply code/game/dna/genes/goon_powers.dm]: why not

**Pull request merged**